openedx/core/djangoapps/catalog/utils.py

```diff
@@ -93,16 +112,118 @@ def get_program_types():
         return []
 
 
-def get_programs_with_type_logo():
+def _get_program_instructors(program):
+    """
+    Returns a list of dicts representing the instructor profile data for each unique
+    instructor associated with a course run in the program. This instructor profile data
+    is stored on the course module. We cache the instructor info to avoid course module
+    lookups on every call.
+
+    Arguments:
+        program (dict): representing the program.
+
+    Returns:
+        list of dict, representing the unique set of instructors for the program.
+    """
+    cache_key = 'program.instructors.{program_id}'.format(
+        program_id=program.get('uuid')
+    )
+
+    program_instructors_dict = {}
+    program_instructors_list = cache.get(cache_key, [])
+    if program_instructors_list:
+        return program_instructors_list
+
+    module_store = modulestore()
+    for course_run_key in _get_program_course_run_keys(program):
+        course_descriptor = module_store.get_course(course_run_key)
+        if course_descriptor:
+            course_instructors = getattr(course_descriptor, 'instructor_info', {})
+            # Deduplicate program instructors using instructor name
+            program_instructors_dict.update(
+                {instructor.get('name'): instructor for instructor in course_instructors.get('instructors', [])}
+            )
+
+    program_instructors_list = program_instructors_dict.values()
+    cache.set(cache_key, program_instructors_list)
+
+    return program_instructors_list
+
+
+def _get_program_course_run_keys(program):
     """
-    Join program type logos with programs of corresponding type.
+    Returns a list of course keys for each course run in the given program.
+
+    Arguments:
+        program (dict): representing the program.
+
+    Returns:
+        list of CourseKey, for each course run key in the program.
     """
-    programs_list = get_programs()
-    program_types = get_program_types()
+    keys = []
+    for course in program.get('courses', []):  # pylint: disable=E1101
+        for course_run in course.get('course_runs', []):
+            keys.append(CourseKey.from_string(course_run.get('key')))
+    return keys
 
-    type_logo_map = {program_type['name']: program_type['logo_image'] for program_type in program_types}
 
-    for program in programs_list:
-        program['logo_image'] = type_logo_map[program['type']]
+def get_program_with_type_and_instructors(marketing_slug):
```

Comment:
I'd like to dig deeper here. We use the UUID to construct URLs for the existing LMS program detail pages. The `marketing_slug` field is a holdover from the old programs service, and it's actually unused now. Can you use the UUID instead?